### PR TITLE
Frontend: Created the Homepage with the hamburger menu

### PR DESCRIPTION
**Pull request by sksmagr23**
## Description
I Created the home page with hamburger menu based on the given figma design. I added created the menu in Navbar.tsx and import this in home.tsx(landing page). I also added the necessary media in assets/src and also slightly modified my previously button component, also added some custom classes in tailwind.config.js. I showed just for instance the sidebar after login, which will be later implemented.

Fixes: #78

## How to Test
- Navigate to frontend directory
- Run "npm run dev"

## Related Issues
- #78 

## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have assigned reviewers to this pull request.

## Screenshots
Screenshot from 2024-06-10 11-06-34
Screenshot from 2024-06-10 11-06-42
Screenshot from 2024-06-10 11-06-47
Screenshot from 2024-06-10 11-07-34

### CODE REVIEW

**Comment by kuv2707** [comment]
We should add an AppContext handling user signin and use it here to sync the UI with logged in state.
I'll do it. In the meanwhile, @shivansh-bhatnagar18 can review the design and functionality

**Comment by shivansh-bhatnagar18** [comment]
@kuv2707 The PR looks good to me. Its just that the images are not getting rendered.

**Comment by sksmagr23** [comment]
Bhaiya it might be the loading error, because it is opening in my localhost

File: frontend/src/Navbar.tsx [hunk 1]
```diff
@@ -1,7 +1,113 @@
...
+        setIsOpen(!isOpen);
+    };
+
+    return (
+        <div>
+            <div className="flex justify-between items-center p-4 ml-4 mt-4 mr-2 relative z-10">
+                <div className="space-x-4">
+                    <button onClick={toggleMenu}>
+                        <img
+                            src="src/assets/hamburger.png"
```
Comment (kuv2707) [inline frontend/src/Navbar.tsx:30]: Try removing `src` from the src attribute (include the /)

If that doesn't work, rename "assets" to "public" and then write src as "/hamburger.png".

File: frontend/src/Navbar.tsx [hunk 2]
```diff
@@ -1,7 +1,113 @@
...
+            </div>
+            {isOpen && (
+                <div className="fixed inset-0 flex z-20">
+                    <div className="bg-gray-300 text-black border-gray-600 w-64 p-4 shadow-lg pl-10 rounded-r-lg relative z-30">
+                        <button
+                            onClick={toggleMenu}
+                            className="absolute top-3 right-3"
+                        >
+                            <img
+                                src="src/assets/close.png"
```
Comment (kuv2707) [inline frontend/src/Navbar.tsx:53]: here too

File: frontend/src/Navbar.tsx [hunk 3]
```diff
@@ -1,7 +1,113 @@
...
-}
+type NavbarProps = {
+    isLoggedIn?: boolean;
+    username?: string;
+    onLogin?: () => void;
+    onLogout?: () => void;
+};
+
+const Navbar: React.FC<NavbarProps> = ({
+    isLoggedIn,
```
Comment (kuv2707) [inline frontend/src/Navbar.tsx:13]: These will be accessed via the context, which I'll add when I get time.

**Review by kuv2707 — COMMENTED** [review]
Posted some comments

File: frontend/src/Navbar.tsx (hunk 3 above)
Comment (sksmagr23) [reply frontend/src/Navbar.tsx:13]: So what I do here

**Pull request merged**